### PR TITLE
DBZ-699 Handle all exceptions

### PR DESCRIPTION
**Pull request by jpechane**

### CODE REVIEW

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java [hunk 1]
```diff
@@ -291,14 +290,18 @@ private PGReplicationStream startPgReplicationStream(final LogSequenceNumber lsn
     public synchronized void close() {
         try {
             super.close();
-        } catch (SQLException e) {
+        } catch (Exception e) {
```
Comment (jchipmunk) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java]: I think catching `Throwable` is better. I found many examples of this approach in Debezium project.

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/connection/ReplicationConnectionIT.java [hunk 2]
```diff
@@ -61,7 +62,7 @@ public void shouldCreateAndDropReplicationSlots() throws Exception {
         }
     }
```
Comment (jchipmunk) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/connection/ReplicationConnectionIT.java]: I think you need to add test that the connection is closed if the replication slot name is invalid (for example: it contains `-` character)

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java]: I am not a fan of catching Throwables but in resource cleanup method it is justified.

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java [hunk 3]
```diff
@@ -291,14 +290,18 @@ private PGReplicationStream startPgReplicationStream(final LogSequenceNumber lsn
     public synchronized void close() {
         try {
             super.close();
-        } catch (SQLException e) {
+        } catch (Throwable e) {
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java]: No code after "}" please :) Sorry for the nitpicking, but the quicker we all adjust to the agreed-on style, the less friction down the road. Thanks!

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java [hunk 4]
```diff
@@ -291,14 +290,18 @@ private PGReplicationStream startPgReplicationStream(final LogSequenceNumber lsn
...
-            try (PostgresConnection connection = new PostgresConnection(originalConfig)) {
-                connection.dropReplicationSlot(slotName);
+        try {
+            if (dropSlotOnClose) {
+                // we're dropping the replication slot via a regular - i.e. not a replication - connection
+                try (PostgresConnection connection = new PostgresConnection(originalConfig)) {
+                    connection.dropReplicationSlot(slotName);
+                }
             }
+        } catch (Throwable e) {
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java]: Same as above.

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java [hunk 5]
```diff
@@ -291,14 +290,18 @@ private PGReplicationStream startPgReplicationStream(final LogSequenceNumber lsn
...
             super.close();
-        } catch (SQLException e) {
+        } catch (Throwable e) {
             LOGGER.error("Unexpected error while closing Postgres connection", e);
         }
-        if (dropSlotOnClose) {
-            // we're dropping the replication slot via a regular - i.e. not a replication - connection
-            try (PostgresConnection connection = new PostgresConnection(originalConfig)) {
-                connection.dropReplicationSlot(slotName);
+        try {
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java]: Seems simpler to add the catch to the inner try(-with-resources)?

Comment (jpechane) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java]: @gunnarmorling Right!

**Comment by gunnarmorling** [comment]
Applied. Thanks, @jpechane!

**Pull request merged**